staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion_test.go

```diff
@@ -0,0 +1,544 @@
+		t.Run(version, func(t *testing.T) {
+			name := "storageversion-" + version
+			client := ctc.versionedClient(ns, version)
+			obj, err := client.Create(NewMultiVersionInstance(ns, name, version), metav1.CreateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			ctc.setStorageVersion(t, "v1beta2")
+			ctc.waitForStorageVersion(t, "v1beta2", client, obj)
+
+			obj, err = client.Get(obj.GetName(), metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ctc.setStorageVersion(t, "v1beta1")
+			ctc.waitForStorageVersion(t, "v1beta1", client, obj)
+		})
+	}
+}
+
+func validateMixedStorageVersions(t *testing.T, ctc *conversionTestContext) {
+	ns := ctc.namespace
+	defer ctc.setStorageVersion(t, "v1beta2")
+
+	v1client := ctc.versionedClient(ns, "v1beta1")
+	v2client := ctc.versionedClient(ns, "v1beta2")
+	clients := map[string]dynamic.ResourceInterface{"v1beta1": v1client, "v1beta2": v2client}
+	versions := []string{"v1beta1", "v1beta2"}
+	for _, version := range versions {
+		name := "stored-at-" + version
+		obj, err := v1client.Create(NewMultiVersionInstance(ns, name, "v1beta1"), metav1.CreateOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+		ctc.setStorageVersion(t, version)
+		ctc.waitForStorageVersion(t, version, v1client, obj)
+	}
+
+	for clientVersion, client := range clients {
+		t.Run(clientVersion, func(t *testing.T) {
+			objs := []*unstructured.Unstructured{}
+			for _, version := range versions {
+				obj, err := client.Get("stored-at-"+version, metav1.GetOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+				objs = append(objs, obj)
+			}
+			o1 := objs[0].Object
+			for i := 1; i < len(objs); i++ {
+				o2 := objs[i].Object
+				// ignore metadata for comparison purposes
+				delete(o1, "metadata")
+				delete(o2, "metadata")
+				if !reflect.DeepEqual(o1, o2) {
+					t.Errorf("Expected %+v to be the same if identical object stored at version %s but got %+v", o1, versions[i], o2)
+				}
+			}
+		})
+	}
+}
+
+func validateServed(t *testing.T, ctc *conversionTestContext) {
+	ns := ctc.namespace
+
+	for _, version := range []string{"v1beta1", "v1beta2"} {
+		t.Run(version, func(t *testing.T) {
+			name := "served-" + version
+			client := ctc.versionedClient(ns, version)
+			obj, err := client.Create(NewMultiVersionInstance(ns, name, version), metav1.CreateOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+			ctc.setServed(t, version, false)
+			ctc.waitForServed(t, version, false, client, obj)
+			ctc.setServed(t, version, true)
+			ctc.waitForServed(t, version, true, client, obj)
+		})
+	}
+}
+
+func expectConversionFailureMessage(id, message string) func(t *testing.T, ctc *conversionTestContext) {
+	return func(t *testing.T, ctc *conversionTestContext) {
+		ns := ctc.namespace
+		v1client := ctc.versionedClient(ns, "v1beta1")
+		v2client := ctc.versionedClient(ns, "v1beta2")
+		var err error
+		// storage version is v1beta2, so this skips conversion
```

Comment:
set and wait for storage version in case it got changed by other testcase / check

Reply:
Done.